tpm2tools/flush_active_handles.go

```diff
@@ -0,0 +1,40 @@
+package tpm2tools
+
+
+import (
+    "io"
+    "github.com/google/go-tpm/tpm2"
+    "github.com/google/go-tpm/tpmutil"
+)
+
+// FlushActiveHandles calls FlushContext() on all active handles within the 
+// TPM at io.ReadWriter rw. Returns nil if successful or TPM has no active
+// handles.
+func FlushActiveHandles(rw io.ReadWriter) error {
+    handles, err := activeHandles(rw)
+    if err != nil {
+        return err
+    }
+    for _, handle := range handles {
+        err = tpm2.FlushContext(rw, handle)
+        if err != nil {
+            return err
+        }
+    }
+    return nil
+}
+
+// activeHandles returns a slice of tpmutil.Handle objects of all handles
+// active within the TPM at ioReadWriter rw.
+func activeHandles(rw io.ReadWriter) ([]tpmutil.Handle, error) {
+    vals, _, err := tpm2.GetCapability(rw, tpm2.CapabilityHandles, ^uint32(0), 0x80000000)
```

Comment:
Here we are passing ^uint32(0) as the count parameter. We should instead pass MAX_CAP_HANDLES (a constant that might need to be defined in this library. See section 5.5 of [part 4 of the spec](https://trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-4-Supporting-Routines-01.38-code.pdf)

Reply:
Specifically, we would want to define (at least) the following constants:

```go
const (
	CapBufferLenMax uint32 = 1024
	CapDataMax      uint32 = CapBufferLenMax - 4 - 4
	CapHandlesMax   uint32 = CapDataMax / 4
)
```

@awly, should these be in tpm2/constants.go or this library?

Reply:
Since these directly map to constants from the spec, they should go in tpm2/constants.go.

Reply:
Where does the `CapBufferLenMax` of `1024` come from? I am adding this to tpm2/constants.go but following the documentation it isn't clear what this number is. Does it vary from TPM to TPM?

Reply:
I was looking at the note on `MAX_CAP_BUFFER` in Section 10.10 of [Part 2 of the spec](https://trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf).

However, while a value of 1024 seems to be typical, the spec never actually says it must be that value. The code in part 4 of the spec is merely "illustrative". However, looking at the other code for `GetCapability` it doesn't seem like TPM2 has support for a `MAX_CAP_BUFFER` bigger than the [TPM's max response size](https://github.com/google/go-tpm/blob/79047b3352714e31d8b648e8f2d783c4a88db580/tpmutil/run.go#L31).

So I think it's best we just pass [MaxUint32](https://golang.org/pkg/math/#pkg-constants) for the number of handles, and just make sure `moredata` is false. Sorry, I thought MAX_CAP_BUFFER was a fixed constant.

Reply:
Done.